docs/contributing.md

```diff
@@ -0,0 +1,203 @@
+# Contributing
+
+Thanks for considering contributing to HTTP Core!
+
+We welcome contributors to:
+
+- Try [HTTPX](https://www.python-httpx.org), as it is HTTP Core's main entry point,
+and [report bugs/issues you find](https://github.com/encode/httpx/issues/new)
+- Help triage [issues](https://github.com/encode/httpcore/issues) and investigate
+root causes of bugs
+- [Review Pull Requests of others](https://github.com/encode/httpcore/pulls)
+- Review, clarify and write documentation
+- Participate in discussions
+
+## Reporting Bugs or Other Issues
+
+HTTP Core is a fairly specialized library and its main purpose is to provide a
+solid base for [HTTPX](https://www.python-httpx.org). HTTPX should be considered
+the main entry point to HTTP Core and as such we encourage users to test and raise
+issues in [HTTPX's issue tracker](https://github.com/encode/httpx/issues/new)
+where maintainers and contributors can triage and move to HTTP Core if appropriate.
+
+If you are convinced that the cause of the issue is on HTTP Core you're more than
+welcome to [open an issue](https://github.com/encode/httpcore/issues/new).
+
+Please attach as much detail as possible and, in case of a
+bug report, provide information like:
+
+- OS platform or Docker image
+- Python version
+- Installed dependencies and versions (`python -m pip freeze`)
+- Code snippet to reproduce the issue
+- Error traceback and output
```

Comment:
Okay with this, though in my experience people tend to actually not look at the contributing guide and forget to include these basic repro info bits. :-) But we can look at setting up issue templates as a follow up, though for now I think the volume is manageable…

Reply:
Sure, HTTPX also has this so I copied over for the most part.